pkg/query/querier.go

```diff
@@ -263,7 +282,22 @@ func storeHintsFromPromHints(hints *storage.SelectHints) *storepb.QueryHints {
 	}
 }
 
+type QuerierWithHints interface {
+	storage.Querier
+	SelectWithHints(context.Context, bool, *storage.SelectHints, ...*labels.Matcher) (storage.SeriesSet, *store.HintsCollector)
+}
+
 func (q *querier) Select(ctx context.Context, _ bool, hints *storage.SelectHints, ms ...*labels.Matcher) storage.SeriesSet {
+	return q.originalSelect(ctx, true, hints, nil, ms...)
+}
+
+func (q *querier) SelectWithHints(ctx context.Context, _ bool, hints *storage.SelectHints, ms ...*labels.Matcher) (storage.SeriesSet, *store.HintsCollector) {
```

Comment:
I dont understand the need for this interface or the method, couldnt select always take hints instead?

Reply:
i mean a hints collector

Reply:
and then we can assume that its always there later

Reply:
We don't want to capture the hints if analysis functionality is disabled. Also, this is an internal Prometheus interface used in a lot of places, in external projects, so it would be cool to have same Select().

Reply:
I'm not sure what's your suggestion. Returning two values instead of one in Select() definitely breaks the Prometheus querier interface. Have you seen https://github.com/thanos-io/promql-engine/pull/325/files? Maybe it will help understand this btter.

Reply:
I missed the different return value! Ok